### DISCUSSION TITLE
How to specify a conversion that only applies to specific MIME types?

### DISCUSSION THREAD

**Discussion by Synchro** [spatie/laravel-medialibrary#2221] [Q&A] [answered]
I'm using my own [`ImageGenerator` for audio files](https://github.com/Synchro/laravel-medialibrary-audio). This works very nicely in Media Library Pro for generating pre-submit thumbnails:

![audio thumbnail](https://user-images.githubusercontent.com/81561/100937645-949dd100-34f3-11eb-85b3-7b935661af04.png)

However, I can't figure out how to make it display this same generated image in other places. If I say `{{ $media }}` for an audio file I get a snippet that does not contain an image tag; I have a `thumb` conversion defined, which works for images, but not for these same audio files, where I would expect it to use the same generator.

Aside from thumbnails, how should I generate different HTML tags for different media types (e.g. as the docs have an example of for video, though it only shows what it might look like, not how to get ML to generate it)?

**Comment by freekmurze** [comment, marked as answer]
When rendering media via Blade tags, [`toHtml`](https://github.com/spatie/laravel-medialibrary/blob/fc3d5500072adaf0b3c8e238517d84b9bb9e0d2b/src/MediaCollections/Models/Media.php#L319) on the Media model will be executed, and one of [these views](https://github.com/spatie/laravel-medialibrary/tree/1c2938a3d056c7de89c3d9bb8f155d553900ed6a/resources/views) will be rendered.

You can customise the output by publishing the view of this package to your project and modifying them there.


```
php artisan vendor:publish --provider="Spatie\MediaLibrary\MediaLibraryServiceProvider" --tag="views"
```
